vscode-lean4/src/utils/clientProvider.ts

```diff
@@ -156,12 +219,10 @@ export class LeanClientProvider implements Disposable {
                 // ignore workspaces that belong to a different version of Lean.
                 console.log(`Lean4 extension ignoring workspace '${folderUri}' because it is not a Lean 4 workspace.`);
                 this.pending.delete(path);
+                this.clients.delete(path);
```

Comment:
Should we dispose the client object in this case?

Reply:
Good catch, fixed.